### ISSUE TITLE
Can't Bind a Custom IBindingList Component to Controls at DesignTime in .NET 8

### ISSUE THREAD

**Issue by AkylzhanDevExpress** [dotnet/winforms#11646]
### Environment

Microsoft Visual Studio Professional 2022 (64-bit) - Current
Version 17.10.3

### .NET version

.NET 8

### Did this work in a previous version of Visual Studio and/or previous .NET release?

Works in .NET Framework

### Issue description

We have a custom component that inherits from IBindingList, ITypedList and we can't bind it to controls at DesignTime. The issue is that it works as expected in .NET Framework

This is how it looks like in Framework
![framework](https://github.com/dotnet/winforms/assets/175200230/0812b473-81b3-42af-9298-a56f0a73769d)

This is in .NET Core
![core](https://github.com/dotnet/winforms/assets/175200230/2a61fc01-161b-4799-b816-d10d4674c56e)

I am creating this issue on behalf of the WinForms team at DevExpress, a Microsoft partner. Our users have encountered the [issue in one of our controls](https://supportcenter.devexpress.com/ticket/details/t1241812/textedit-cannot-bind-to-an-xpservercollectionsource-or-choose-its-properties)

### Steps to reproduce

1. Create a custom component that inherits IBindingList, ITypedList
2. Try binding it to TextBox

Minimal reproduction project attached below
[mrp.zip](https://github.com/user-attachments/files/16158382/mrp.zip)

**Comment by dennis-garavsky** [comment]
@merriemcgaw @Shyam-Gupta JFYI

This appears to be a new behavior introduced in recent Visual Studio versions, because it all worked fine for our custom data source components in the past. For instance, our own ORM and other components have collections like XPCollection, XPServerCollectionSource, etc.). Now it appears that if a collection does not inherit from the standard BindingSource, then it is now impossible to select fields of a data source. Please let us know if this breaking change is intended or if there are ways around for such scenarios with custom collections. Thanks.

![image](https://github.com/dotnet/winforms/assets/5479762/db9ec9a9-868e-418b-845a-37f1e102bcd3)

**Comment by elachlan** [comment]
@Olina-Zhang can your team please investigate?

**Comment by merriemcgaw** [comment]
I'll have @KlausLoeffelmann take a first look. @Olina-Zhang can you repro this and find where it changed, and then copy this issue into the designer repo? @dennis-garavsky we'll keep you posted on our investigation!

**Comment by Olina-Zhang** [comment]
Hi @merriemcgaw @KlausLoeffelmann verified it is not a regression issue, cannot see testdata: Name and Surname under Form1 List Instance.
And we can bind that custom IBindingList Component to control through "Add and update object data source" window, see following screenshot:
![image](https://github.com/dotnet/winforms/assets/26474449/a4f2bf09-fab2-4d72-b0d3-037449601177)

**Comment by JeremyKuhne** [comment]
@KlausLoeffelmann this looks like something we've not got working in Core yet?

**Comment by KlausLoeffelmann** [comment]
Yes. This needs to be addressed completely in the Designer, though, it's not a runtime issue.
@Olina-Zhang: If we haven't yet, can we get that into the Designer repo for VSNext?
Let's keep that open for tracking, but it should be .NET 10 timeframe.

**Comment by Olina-Zhang** [comment]
Filed it WinForms designer repo: https://github.com/microsoft/winforms-designer/issues/6027.
